platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/SignedStateFileReader.java

```diff
@@ -92,7 +97,7 @@ record StateFileData(MerkleRoot state, Hash hash, SigSet sigSet) {}
 
                     final Path directory = stateFile.getParent();
 
-                    final MerkleRoot state = in.readMerkleTree(directory, MAX_MERKLE_NODES_IN_STATE);
+                    final MerkleRoot state = snapshotStateReader.apply(in, directory);
```

Comment:
While I like that this now throws an IOE it has no knowledge about stateFile and therefore the message of the exception can be perfect to describe the technical reason but will not contain the concreate file. Can we add a try/catch around and throw `new IOException("Error reading stateFile " + stateFile.NoIdeHowToGetAnIdentifier(), cause);`?

Reply:
Done